pytorch_lightning/trainer/trainer.py

```diff
@@ -860,14 +830,50 @@ def run_sanity_check(self, ref_model):
             # prevents sanity check to affect random sampling in training
             reset_seed()
 
+    def fit(
```

Comment:
don't we want to move this up? It would be nice if the entry points fit, validate, test came first.

Reply:
Agree. Let's try to organise, so it is simple to follow.

Reply:
I'll do this in a separate PR. Keeping them where they are for now so they are together